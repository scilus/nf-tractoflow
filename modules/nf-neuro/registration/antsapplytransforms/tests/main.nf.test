nextflow_process {

    name "Test Process REGISTRATION_ANTSAPPLYTRANSFORMS"
    script "../main.nf"
    process "REGISTRATION_ANTSAPPLYTRANSFORMS"

    tag "modules"
    tag "modules_nfcore"
    tag "registration"
    tag "registration/antsapplytransforms"

    tag "subworkflows"
    tag "subworkflows/load_test_data"

    setup {
            run("LOAD_TEST_DATA", alias: "LOAD_DATA") {
                script "../../../../../subworkflows/nf-neuro/load_test_data/main.nf"
                process {
                    """
                    input[0] = Channel.from( [ "registration.zip" ] )
                    input[1] = "test.load-test-data"
                    """
                }
            }
        }

    test("registration - antsapplytransforms") {

        config "./nextflow.config"

        when {
            process {
                """
                input[0] = LOAD_DATA.out.test_data_directory
                    .map{ test_data_directory -> [
                        [ id:'test', single_end:false ], // meta map
                        file("\${test_data_directory}/b0.nii.gz"),
                        file("\${test_data_directory}/mni_masked_2x2x2.nii.gz"),
                        file("\${test_data_directory}/output1Warp.nii.gz"),
                        file("\${test_data_directory}/output0GenericAffine.mat")
                    ]}
                """
            }
        }
        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }

    test("registration - antsapplytransforms - stub-run") {

        options "-stub-run"

        config "./nextflow.config"

        when {
            process {
                """
                input[0] = LOAD_DATA.out.test_data_directory
                    .map{ test_data_directory -> [
                        [ id:'test', single_end:false ], // meta map
                        file("\${test_data_directory}/b0.nii.gz"),
                        file("\${test_data_directory}/mni_masked_2x2x2.nii.gz"),
                        file("\${test_data_directory}/output1Warp.nii.gz"),
                        file("\${test_data_directory}/output0GenericAffine.mat")
                    ]}
                """
            }
        }
        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out.versions).match() }
            )
        }
    }
}
