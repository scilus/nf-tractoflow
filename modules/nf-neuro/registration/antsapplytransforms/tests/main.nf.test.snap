{
    "registration - antsapplytransforms": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test__b0__warped.nii.gz:md5,3ae49aacab66fb2378da92155453b1be"
                    ]
                ],
                "1": [
                    "versions.yml:md5,180609e91d7d492cd7db9b09f47f055d"
                ],
                "versions": [
                    "versions.yml:md5,180609e91d7d492cd7db9b09f47f055d"
                ],
                "warped_image": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test__b0__warped.nii.gz:md5,3ae49aacab66fb2378da92155453b1be"
                    ]
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-11-28T20:21:53.029255208"
    },
    "registration - antsapplytransforms - stub-run": {
        "content": [
            [
                "versions.yml:md5,180609e91d7d492cd7db9b09f47f055d"
            ]
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-11-28T20:21:57.642570655"
    }
}